File: internal/output/output_test.go
package output

import (
	"testing"
)

func TestIsJSONMode_ForceJSON(t *testing.T) {
	ForceJSON = true
	defer func() { ForceJSON = false }()

	if !IsJSONMode() {
		t.Error("expected JSON mode when ForceJSON is true")
	}
}

func TestIsJSONMode_Default(t *testing.T) {
	ForceJSON = false
	// In tests, stdout is not a terminal, so this should return true
	if !IsJSONMode() {
		t.Error("expected JSON mode when stdout is not a terminal")
	}
}

File: src/lib/output.test.ts
import { describe, it, expect, afterEach, vi } from "vitest";
import { isJSONMode, setFlags, debug, verbose, debugMode } from "./output.js";

describe("isJSONMode", () => {
  afterEach(() => {
    setFlags({});
  });

  it("returns true when forceJSON is set", () => {
    setFlags({ json: true });
    expect(isJSONMode()).toBe(true);
  });

  it("returns true when stdout is not a TTY (test environment)", () => {
    setFlags({});
    // In tests, stdout is not a terminal
    expect(isJSONMode()).toBe(true);
  });
});

describe("output flags", () => {
  afterEach(() => {
    setFlags({});
  });

  it("tracks verbose and debug independently", () => {
    setFlags({ verbose: true, debug: false });
    expect(verbose).toBe(true);
    expect(debugMode).toBe(false);

    setFlags({ verbose: false, debug: true });
    expect(verbose).toBe(false);
    expect(debugMode).toBe(true);
  });

  it("only prints debug logs when debug mode is enabled", () => {
    const errSpy = vi.spyOn(console, "error").mockImplementation(() => {});

    setFlags({ debug: false });
    debug("quiet");
    expect(errSpy).not.toHaveBeenCalled();

    setFlags({ debug: true });
    debug("loud");
    expect(errSpy).toHaveBeenCalledWith("[debug] loud");

    errSpy.mockRestore();
  });
});
